packages/fetchai/skills/tac_negotiation/handlers.py

```diff
@@ -267,32 +274,66 @@ def _on_accept(self, accept: FipaMessage, fipa_dialogue: FipaDialogue) -> None:
-                    counterparty=LEDGER_API_ADDRESS,
-                    performative=ContractApiMessage.Performative.GET_RAW_MESSAGE,
-                    ledger_id=strategy.ledger_id,
-                    contract_id=strategy.contract_id,
-                    contract_address=strategy.contract_address,
-                    callable="get_hash_batch",
-                    kwargs=ContractApiMessage.Kwargs(
-                        strategy.kwargs_from_terms(
-                            fipa_dialogue.terms, is_from_terms_sender=False
-                        )
-                    ),
-                )
-                contract_api_dialogue = cast(ContractApiDialogue, contract_api_dialogue)
-                contract_api_dialogue.associated_fipa_dialogue = fipa_dialogue
-                self.context.logger.info(
-                    "requesting batch transaction hash, sending {} to {}, message={}".format(
-                        contract_api_msg.performative,
-                        strategy.contract_id,
+                if strategy.ledger_id == EthereumApi.identifier:
+                    contract_api_dialogues = cast(
+                        ContractApiDialogues, self.context.contract_api_dialogues
+                    )
+                    (
                         contract_api_msg,
+                        contract_api_dialogue,
+                    ) = contract_api_dialogues.create(
+                        counterparty=LEDGER_API_ADDRESS,
+                        performative=ContractApiMessage.Performative.GET_RAW_MESSAGE,
+                        ledger_id=strategy.ledger_id,
+                        contract_id=strategy.contract_id,
+                        contract_address=strategy.contract_address,
+                        callable="get_hash_batch",
+                        kwargs=ContractApiMessage.Kwargs(
+                            strategy.kwargs_from_terms(
+                                fipa_dialogue.terms, is_from_terms_sender=False
+                            )
+                        ),
                     )
-                )
-                self.context.outbox.put_message(message=contract_api_msg)
+                    contract_api_dialogue = cast(
+                        ContractApiDialogue, contract_api_dialogue
+                    )
+                    contract_api_dialogue.associated_fipa_dialogue = fipa_dialogue
+                    self.context.logger.info(
+                        "requesting batch transaction hash, sending {} to {}, message={}".format(
+                            contract_api_msg.performative,
+                            strategy.contract_id,
+                            contract_api_msg,
+                        )
+                    )
+                    self.context.outbox.put_message(message=contract_api_msg)
+                elif strategy.ledger_id == FetchAIApi.identifier:
+                    public_keys = self.context.public_keys
+                    if public_keys is None:
```

Comment:
Why would it ever be None? That makes no sense. An agent by design needs to have a wallet otherwise it won't run. See also above, the public_keys dictionary is not optional.